crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -210,12 +213,16 @@ pub async fn sync(
 
         tx_event
             .send(Event::Update(
-                (block, commitments),
+                (block.clone(), commitments),
                 Box::new(state_update),
                 timings,
             ))
             .await
             .context("Event channel closed")?;
+
+        websocket_txs
+            .new_head
+            .send_if_receiving(WebsocketEventNewHead::new(*block.clone()))
```

Comment:
The second clone unnecessarily clones the transactions and the receipts. You could remove all the clones by making `WebsocketEventNewHead` take a ref to a block in ctor (or better `impl From<&Block>`), and then:
1. create head from &block
2. send the `Update` event which would consume the block
3. send the head